gr-audio/lib/CMakeLists.txt

```diff
@@ -44,7 +44,7 @@ endif(ENABLE_GR_CTRLPORT)
 ########################################################################
 find_package(ALSA)
 
-if(ALSA_FOUND)
+if((ALSA_FOUND) AND NOT (CMAKE_SYSTEM_NAME STREQUAL "kFreeBSD") AND NOT (CMAKE_SYSTEM_NAME STREQUAL GNU))
```

Comment:
Not sure about the GNU part again? Is this correct? Will this disable ALSA on all GNU platforms?

Reply:
No, it will not build ALSA blocks on GNU Hurd

Reply:
Ah, then only Linux, MacOS and "normal" FreeBSD will build ALSA blocks. Why not kFreeBSD & GNU Hurd if ALSA is available?

Reply:
AFAIK GNU Hurd simply has no proper ALSA: see this *exquisite* documentation [or chat log](https://www.gnu.org/software/hurd/open_issues/anatomy_of_a_hurd_system.html) and debian's own guideline to disable ALSA on hurd: https://www.debian.org/ports/hurd/hurd-devel-debian. 

All in all, it's absolutely not surprising that Hurd doesn't do monolithic syscall-style ALSA. It's a microkernel! Of course,it emulates OSS, but ALSA's simply not there.

kFreeBSD simply doesn't have an ALSA interface. There's salsa to (very imperfectly) emulate ALSA, afaik.

So, to answer the question "why not ALSA if ALSA is available?": It's not available.